### PR TITLE
Update Lightning import error handling (#13376)

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

- Expand `JOB_TYPE` documentation for `ADMIN SHOW DDL...` output.
- Add note about `temp-dir` to `tidb_enabe_fast_reorg` docs.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
Some very useful comments in the code:
https://github.com/pingcap/tidb/blob/275696aa449bd7bff5c85ba3f8013d162dc690a4/parser/model/reorg.go#L37-L54

**Comment by dveeden** [comment]
{"level":"ERROR","time":"2023/04/24 08:46:23.897 +00:00","caller":"backend_mgr.go:66","message":"[ddl-ingest] build lightning backend failed, will use kernel index reorg method to backfill the index","job ID":32858,"error":"[Lightning:Config:ErrInvalidSortedKVDir]invalid sorted-kv-dir '/tmp/tidb/tmp_ddl-3306/32858' for local backend, please change the config or delete the path: mkdir /tmp/tidb/tmp_ddl-3306/32858: no such file or directory"

**Review by tangenta — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -1144,7 +1144,13 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiDB node size is at least 8 vCPU. For [Serverless Tier](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta) clusters, this feature is unavailable.
 
 </CustomContent>
```
Comment (qiancai) [inline system-variables.md:1146]:
```suggestion
</CustomContent>

```

File: system-variables.md [hunk 2]
```diff
@@ -1144,7 +1144,13 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
...
+<CustomContent platform="tidb">
 
+> **Note:**
+>
+> Index accelleration requires a [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) that is writable and has enough free space to be able to function. If the `temp-dir` is unusable TiDB will fall back to non-accelerated index building.
```
Comment (qiancai) [inline system-variables.md:1151]:
```suggestion
> Index acceleration requires a [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) that is writable and has enough free space. If the `temp-dir` is unavailable, TiDB will fall back to non-accelerated index building.
```

File: sql-statements/sql-statement-admin-show-ddl.md [hunk 3]
```diff
@@ -51,7 +51,10 @@ The `ADMIN SHOW DDL JOBS` statement is used to view all the results in the curre
 - `JOB_ID`: each DDL operation corresponds to a DDL job. `JOB_ID` is globally unique.
 - `DB_NAME`: the name of the database where the DDL operation is performed.
 - `TABLE_NAME`: the name of the table where the DDL operation is performed.
-- `JOB_TYPE`: the type of DDL operation.
+- `JOB_TYPE`: the type of DDL operation. Common job types include the following:
+    - `ingest`: Ingestion with accelerated index backfilling as configured by [`tidb_ddl_enable_fast_reorg`](/system-variables.md#tidb_ddl_enable_fast_reorg-new-in-v630).
+    - `txn`: Basic transactional backfill.
+    - `txn-merge`: Transactional backfill with temporary index that gets merged with the original index when finished.
```
Comment (ran-huang) [inline sql-statements/sql-statement-admin-show-ddl.md:57]:
```suggestion
    - `txn-merge`: Transactional backfill with a temporary index that gets merged with the original index when the backfill is finished.
```

**Comment by dveeden** [comment]
@ran-huang @qiancai PTAL

**Review by ran-huang — APPROVED** [review]
LGTM

File: system-variables.md [hunk 4]
```diff
@@ -1145,6 +1145,13 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 </CustomContent>
 
+<CustomContent platform="tidb">
+
+> **Note:**
+>
+> Index acceleration requires a [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) that is writable and has enough free space. If the `temp-dir` is unusable, TiDB falls back to non-accelerated index building.
```
Comment (ran-huang) [inline system-variables.md:1152]: Is this line duplicated with https://github.com/pingcap/docs/pull/13584/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R1165 ?

Comment (dveeden) [reply system-variables.md]: I've updated this a bit. PTAL @ran-huang

File: system-variables.md [hunk 5]
```diff
@@ -1332,6 +1332,10 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 >
 > Before you upgrade TiDB to v6.5.0 or later, it is recommended that you check whether the [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) path of TiDB is correctly mounted to an SSD disk. This path is a TiDB configuration item, which takes effect after TiDB is restarted. Therefore, setting this configuration item before upgrading can avoid another restart.
 
+> **Note:**
+>
+> Index acceleration requires a [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) that is writable and has enough free space. If the `temp-dir` is unusable, TiDB falls back to non-accelerated index building. It is recommended to put the `temp-dir` on a SSD disk.
```
Comment (ran-huang) [inline system-variables.md:1337]: Can you please merge the two notes into one? Like using a list:

> Note:
> - one note...
> - another note...

The rest LGTM

Comment (dveeden) [reply system-variables.md]: Done.

File: system-variables.md [hunk 6]
```diff
@@ -1330,7 +1330,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 > **Note:**
 >
```
Comment (qiancai) [inline system-variables.md:1332]: Can we swap the sequence of the two notes to let users understand what `temp-dir` is and then tell them why the configuration is recommended before the upgrade?

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**